lib/Pipedrive.js

```diff
@@ -28,6 +30,71 @@
 			callback(error, collectionItems, additionalData);
 		}, false);
 	};
+	exports.OAuthAuthenticate = function (client_id, redirect_uri, res) {
+	    res.redirect('https://oauth.pipedrive.com/oauth/authorize?' +
+            querystring.stringify({
+                client_id: client_id,
+                redirect_uri: redirect_uri,
+            }))
+	};
+
+	exports.OAuthAuthenticateCallback = function (req, res, redirect_uri, clientId, clientSecret) {
```

Comment:
clientId and clientSecret could be initializeable param, alternative to API token, because of how global these vars are & how they affect authentication mode

Reply:
We tried to reuse as much of the existing logic as possible to ensure backward compatibility. We modified the functions to accommodate authentication with OAuth, but main logic in the client is left untouched. The authentication flow is performed separately and before the creation of a client. Client is supplemented with the access token when the authentication is already performed. The decision to keep client secret and client id away from Client object was deliberate.